### ISSUE TITLE
Skip feature-gated tests in non-staging/stable Windows test runs

### ISSUE THREAD

**Issue by ddebroy** [kubernetes/kubernetes#75147]
**What happened**:
Today Windows test runs focussing on stable features (such as `ci-kubernetes-e2e-aks-engine-azure-1-14-windows`) are not excluding feature gated tests. This will cause failures in an environment without the necessary feature gates enabled. The typical convention for feature gated tests is mentioned here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md#non-default-tests. 

**What you expected to happen**:
Unless the test cluster is created with various/all feature gates enabled, the test run should skip feature gated tests. This can be achieved by specifying `ginkgo.skip=...|\[Feature:.+\]` as done elsewhere like https://github.com/kubernetes/test-infra/blob/master/config/jobs/gke/containerd/gke-test-containerd.yaml#L33 and most PR validating tests.

**How to reproduce it (as minimally and precisely as possible)**:
Feature gated tests like [GMSA](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/windows/gmsa.go#L33) will fail (unless excluded through some other specific string as done in https://github.com/kubernetes/test-infra/pull/11671)

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
1.14
- Cloud provider or hardware configuration:
all
- OS (e.g: `cat /etc/os-release`):
Windows

/sig windows

**Comment by yujuhong** [comment]
@ddebroy I think this was fixed in test-infra. Closing?

**Comment by ddebroy** [comment]
@yujuhong  do you happen to have a pointer to a PR that skips `[Feature]` tagged tests? 

I think we are still picking up Feature tagged tests like `[sig-api-machinery] ResourceQuota [Feature:PodPriority] *` and `[sig-api-machinery] ResourceQuota [Feature:ScopeSelectors] *` in https://testgrid.k8s.io/sig-windows#aks-engine-azure-windows-master 

Since PodPriority and ScopeSelectors are enabled by default now it seems, we can:
1. Get rid of the feature tags in the PodPriority and ScopeSelectors tests.
2. Get rid of the feature tag `Windows` in the [Memory Limits](https://github.com/kubernetes/kubernetes/blob/5c2fe63591acab508e1ff5771c3aa70268852318/test/e2e/windows/memory_limits.go#L44) and [Density](https://github.com/kubernetes/kubernetes/blob/5c2fe63591acab508e1ff5771c3aa70268852318/test/e2e/windows/density.go#L39) tests.
3. And finally enable `ginkgo.skip=...|\[Feature:.+\]` to skip feature tagged tests?

/cc @adelina-t Does ^ sound okay?

**Comment by yujuhong** [comment]
Hmm...the Azure test job has a pretty long ginkgo.focus string. I didn't notice that before.
```
 --ginkgo.focus=\\[Conformance\\]|\\[NodeConformance\\]|\\[sig-windows\\]|\\[sig-apps\\].CronJob|\\[sig-api-machinery\\].ResourceQuota|\\[sig-scheduling\\].SchedulerPreemption --ginkgo.skip=\\[LinuxOnly\\]|GMSA|\\[Serial\\]"
```

IMO, the main test job should not include "Feature". One of the reasons to add "Feature" is to run it in a separate suite. At least that's how we configured GCE test jobs.
For example, the Density tests should probably best not to be run with other tests simultaneously, so having a separate test suite seems reasonable.

**Comment by adelina-t** [comment]
Ok, so at the moment on Azure we have indeed a pretty long focus string. We have the usual suspects: conformance, nodeConformance and sig-windows as well as some non conformance tests: CronJobs , ResourceQuota and SchedulerPreemtion that were considered relevant to show a fully functioning cluster for the release. We don't explicitly exclude all Feature tests, only the ones that the cluster is not prepared to run (i.e GMSA ).

@ddebroy Basically: `[sig-api-machinery] ResourceQuota [Feature:PodPriority] * and [sig-api-machinery] ResourceQuota [Feature:ScopeSelectors] *` tests are not being picked up as an omission, they were explicitly targeted in the focus regex for the reasons listed above. The Density and Memory Limits tests, yes, they get picked up by sig-windows tag.

To recap:

> Get rid of the feature tags in the PodPriority and ScopeSelectors tests.

Sure, I guess this is a sig-testing decision

> Get rid of the feature tag Windows in the Memory Limits and Density tests.

Yes, that should be done.

> And finally enable ginkgo.skip=...|\[Feature:.+\] to skip feature tagged tests?

Sure, but as I said, in aks-engine test runs we aren't picking them by accident. And with the exceptions I mentioned above, we only run conformance, thus there's no chance of picking unintended Feature tests.
